hummingbot/connector/exchange/bittrex/bittrex_exchange.py

```diff
@@ -0,0 +1,403 @@
+
+    async def _place_cancel(self, order_id: str, tracked_order: InFlightOrder):
+        exchange_id = await tracked_order.get_exchange_order_id()
+        api_params = {
+            "orderId": exchange_id
+        }
+        cancel_result = await self._api_delete(
+            path_url=CONSTANTS.ORDER_DETAIL_URL.format(exchange_id),
+            params=api_params,
+            is_auth_required=True,
+            limit_id=CONSTANTS.ORDER_DETAIL_LIMIT_ID)
+        if cancel_result["status"] == "CLOSED":
+            return True
+        return False
+
+    async def _update_balances(self):
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+        account_balances = await self._api_get(path_url=CONSTANTS.BALANCES_URL, is_auth_required=True)
+        for balance_entry in account_balances:
+            asset_name = balance_entry["currencySymbol"]
+            available_balance = Decimal(balance_entry["available"])
+            total_balance = Decimal(balance_entry["total"])
+            self._account_available_balances[asset_name] = available_balance
+            self._account_balances[asset_name] = total_balance
+            remote_asset_names.add(asset_name)
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+    async def _format_trading_rules(self, markets: List) -> List[TradingRule]:
+        retval = []
+        for market in markets:
+            try:
+                trading_pair = await self.trading_pair_associated_to_exchange_symbol(market.get("symbol"))
+                min_trade_size = market.get("minTradeSize")
+                precision = market.get("precision")
+                retval.append(TradingRule(trading_pair,
+                                          min_order_size=Decimal(min_trade_size),
+                                          min_price_increment=Decimal(f"1e-{precision}"),
+                                          min_base_amount_increment=Decimal(f"1e-{precision}"),
+                                          ))
+            except KeyError:
+                self.logger().error(f"Trading-pair {market['symbol']} is not active. Skipping.")
+                continue
+            except Exception:
+                self.logger().error(f"Error parsing the trading pair rule {market}. Skipping.", exc_info=True)
+        return retval
+
+    async def _update_trading_fees(self):
+        resp = await self._api_get(
+            path_url=CONSTANTS.FEES_URL,
+            is_auth_required=True,
+        )
+        for fees in resp:
+            try:
+                trading_pair = await self.trading_pair_associated_to_exchange_symbol(symbol=fees["marketSymbol"])
+                self._trading_fees[trading_pair] = fees
+            except Exception:
+                continue
```

Comment:
The connector does not seem to be using the trading fees information for anything. Why are we requesting them from the exchange?

Reply:
The connector has provided a separate endpoint for the fees, so I believe it best to have a functioning _update_trading_fees() logic even though it's not used, in case we need it in the future.